src/perf/lib/PerfServer.cpp

```diff
@@ -423,3 +569,141 @@ PerfServer::TcpSendCompleteCallback(
         Server->TcpSendDataAllocator.Free(Data);
     }
 }
+
+DelayWorker::DelayWorker() : WakeEvent(false), DoneEvent(true), Lock(), Thread(true)
+{
+}
+
+DelayWorker::~DelayWorker()
+{
+    CXPLAT_FRE_ASSERT(!WorkItems);
+    CXPLAT_FRE_ASSERT(!Initialized);
+}
+
+bool DelayWorker::Initialize(PerfServer* Server, uint16_t PartitionIndex)
+{
+    m_Server = Server;
+    ExecutionContext.Callback = DelayedWork;
+    ExecutionContext.Context = this;
+    InterlockedFetchAndSetBoolean(&ExecutionContext.Ready);
+    ExecutionContext.NextTimeUs = UINT64_MAX;
+
+    uint16_t ThreadFlags = CXPLAT_THREAD_FLAG_SET_IDEAL_PROC;
+    //
+    // Not using the high priority flag for delay threads
+    //
+    CXPLAT_THREAD_CONFIG Config = { ThreadFlags, PartitionIndex, "DelayWorker", WorkerThread, this };
+    if (QUIC_FAILED(Thread.Create(&Config))) {
+        WriteOutput("CxPlatThreadCreate FAILED\n");
+        return false;
+    }
+    Initialized = true;
+    return true;
+}
+
+void DelayWorker::Shutdown()
+{
+    Shuttingdown = true;
+
+    if (Initialized) {
+        WakeWorkerThread();
+        Initialized = false;
+        //
+        // delete any pending work items
+        //
+        Lock.Acquire();
+        DelayedWorkContext* CurrentWorkItem = WorkItems;
+        while (nullptr != CurrentWorkItem) {
+            DelayedWorkContext* NextWorkItem;
+            if (WorkItemsTail == &CurrentWorkItem->Next) {
+                NextWorkItem = nullptr;
+            }
+            else {
+                NextWorkItem = CurrentWorkItem->Next;
+            }
+            delete CurrentWorkItem;
+            CurrentWorkItem = NextWorkItem;
+        }
+        WorkItems = nullptr;
+        Lock.Release();
+    }
+}
+
+void DelayWorker::WakeWorkerThread() {
+    if (!InterlockedFetchAndSetBoolean(&ExecutionContext.Ready)) {
+            CxPlatEventSet(WakeEvent);
+    }
+}
+
+CXPLAT_THREAD_CALLBACK(DelayWorker::WorkerThread, Context)
+{
+    DelayWorker* This = (DelayWorker*)Context;
+    CXPLAT_EXECUTION_STATE DummyState = {
```

Comment:
You're never using `CXPLAT_EXECUTION_STATE` anywhere. Go ahead and remove all references.